docs/apps/build-system.adoc

```diff
@@ -89,33 +89,51 @@ NOTE:  XP apps deployed via file (locally) are presented with a small blue icon
 
 === Development Mode
 
-Enonic XP supports a so-called Development mode.
-When running in this mode, XP will automatically source JavaScript controllers directly from your project folder.
+[WARNING]
+====
+As the name implies, development mode is intended only for development use and should _not_ be used in production. Doing so would cause both security and performance issues.
+====
 
-This may be convenient if you are developing with pure JavaScript and do not depend on build steps in your project.
+Enonic XP can be run in a special mode known as _development mode_. In this mode, XP will detect changes to your project source files, including JavaScript controllers and registered XML schemas. It does this by reading these files directly from their source locations instead of from the application `.jar` file.
 
-You may activate dev mode when starting XP as follows:
+[NOTE]
+====
+Because of the way that XP works, some changes are not picked up in dev mode: if you add a _new_ content type, XP will not pick it up automatically. This is because content types are registered with XP when _deploying_ an app. However, changes to existing content types _are_ detected. If you'd like to have new content types picked up automatically, try using xref:build-system#_continuous_building[continuous build mode].
```

Comment:
> This is because content types are registered with XP when _deploying_ an app

I'd rather take a look at it. Application loader code is complicated and there might be a bug, but I can't confirm that content types get registered when app is deployed.

Reply:
Aight, got it. I'll look into rewording this next week. It's not really important *why* it doesn't work, as long as the reader knows that it doesn't. Thanks!